multinet/api.py

```diff
@@ -35,11 +37,13 @@ def get_workspace(workspace: str) -> Any:
 
 
 @bp.route("/workspaces/<workspace>/tables", methods=["GET"])
-@use_kwargs({"type": fields.Str()})
+@use_args({"type": fields.Str()})
 @swag_from("swagger/workspace_tables.yaml")
-def get_workspace_tables(workspace: str, type: TableType = "all") -> Any:
+def get_workspace_tables(args: Dict, workspace: str) -> Any:
     """Retrieve the tables of a single workspace."""
-    tables = db.workspace_tables(workspace, type)
+    # TODO: Fix mypy complaint about below expression.
+    table_type: TableType = args.get("type") or "all"  # type: ignore
```

Comment:
I'm actually not seeing a mypy error/warning on this line. Do you have more info?

Reply:
If I remove the `# type: ignore` I get the following error:
```
multinet/api.py:45: error: Incompatible types in assignment (expression has type "Union[Any, str]", variable has type "Union[Literal['all'], Literal['node'], Literal['edge']]")
```

Reply:
I'm guessing it's because it has no way of knowing what is assigned to `args["type"]`. However if we remove this change this error will go away

Reply:
Just for future reference: you need to tell mypy that `args.get("type")` will be "restricted" to the type it is expecting. Typescript has a specific way to do this, where you write a bool function whose purpose is to test a value and return `true` if it really is of the correct type, and then Typescript knows enough about that to mark the value as having the restricted type. I'm sure Python has a way to express this though I've never done it and I'm not sure what it's called in Pythonworld.

Also: I am somewhat concerned that I wasn't able to get mypy to raise this error locally. I think it could have to do with those stars in Pipfile maybe. But it's setting off my spider-sense.

Reply:
I had that same thought too but I'm not sure how to do it in python. Might be good to look into.

Did you re-run `pipenv install`? If it's installing from the lockfile it should install the same versions. Either way we should probably pin those.

Reply:
> Did you re-run `pipenv install`? If it's installing from the lockfile it should install the same versions. Either way we should probably pin those.

Whoops, this was the problem. Good catch.